app/controllers/giveaways_controller.rb

```diff
@@ -1,19 +1,36 @@
 class GiveawaysController < ApplicationController
```

Comment:
A standard action is missing.

Reply:
Thanks for pointing out! I missed the edit action definition. I will add it!